var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -200,7 +200,7 @@ class Mfem(Package, CudaPackage, ROCmPackage):
     depends_on('mpi', when='+mpi')
     depends_on('hypre@2.10.0:2.13.99', when='@:3.3.99+mpi')
     depends_on('hypre@:2.20.0', when='@3.4:4.2.99+mpi')
-    depends_on('hypre', when='@4.3.0:+mpi')
+    depends_on('hypre@:2.22.0', when='@4.3.0:+mpi')
```

Comment:
This should be:
```python
    depends_on('hypre@:2.22.0', when='@4.3.0+mpi')
    depends_on('hypre', when='+mpi')
```
In particular, `mfem@develop` should work with new versions of hypre.

Reply:
That is right. I make a commit for that, thanks.

Reply:
@cessenat it looks like this still needs to be updated

Reply:
Sorry @scheibelp but I altered your advice from 

```
    depends_on('hypre@2.10.0:2.13.99', when='@:3.3.99+mpi')
    depends_on('hypre@:2.20.0', when='@3.4:4.3.0+mpi')
```
to
 ```
    depends_on('hypre@2.10.0:2.13.99', when='@:3.3.99+mpi')
    depends_on('hypre@:2.20.0', when='@3.4:4.2.99+mpi')
    depends_on('hypre@:2.22.0', when='@4.3.0:+mpi')
    depends_on('hypre', when='+mpi')
```
Tell me if I made another mistake maybe misusing git...

Reply:
Oh very sorry @scheibelp you were right, 4.3.0:+mpi was wrong, it had to be 4.3.0+mpi.
Corrected at last, and apologize for the waste of time.

Reply:
No problem these things can be easy to miss. Thanks for updating it!